### PR TITLE
Create custom hook

### PR DESCRIPTION
**Pull request by AndrewSakovich**

### CODE REVIEW

File: src/screens/AddNewItemScreen/hooks/usePress.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  const onPressAdd = () => {
+    const newItem = createNewItemHelper(editText, editDate);
+    createNotificationHelper({ channelId, newItem, date: editDate });
+    setLoading(true);
+    dispatch(addItemSagaAction({ newItem, userToken, callback }));
+  };
+
+  const onPress = () => {
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/hooks/usePress.ts]: make it shorter

File: src/screens/AddNewItemScreen/hooks/usePress.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import { AddNewItemScreenNavigationProps } from '../type';
+import { userTokenSelector } from '../../../redux/selectors/userTokenSelector';
+
+export type UsePressParams = {
+  isEdit: boolean;
+  editItem: TodoItemType;
+  editText: TodoItemType['text'];
+  editDate: Date;
+};
+export const usePress = (params: UsePressParams) => {
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/hooks/usePress.ts]: create type for whole hook (with params and return array)

File: src/screens/AddNewItemScreen/hooks/usePress.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const { isEdit, editText, editItem, editDate } = params;
+
+  const [isLoading, setLoading] = useState<boolean>();
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/hooks/usePress.ts]: set initial value and remove type

File: src/screens/AddNewItemScreen/index.tsx [hunk 4]
```diff
@@ -1,87 +1,66 @@
...
-        message:
-          'You have unsaved changes. Are you sure to discard them and leave the screen?',
-        confirmButtonTitle: 'Discard',
-        cancelButtonTitle: "Don't leave",
-      });
-    }
-
-    return back();
-  }, [hasUnsavedChanges]);
+  const onPressBack = usePressBack(hasUnsavedChanges);
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/index.tsx]: move this hook to `useAddNewItemScreen`

File: src/screens/AddNewItemScreen/hooks/usePressBack.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { useCallback } from 'react';
+import { createAlertMessageHelper } from '../../../helpers/createAlertMessageHelper';
+import { useNavigation } from '@react-navigation/native';
+import { AddNewItemScreenNavigationProps } from '../type';
+
+export const usePressBack = (hasUnsavedChanges: boolean) => {
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/hooks/usePressBack.ts]: the same as `usePress`

File: src/screens/AddNewItemScreen/hooks/usePressBack.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  const navigation = useNavigation<AddNewItemScreenNavigationProps>();
+  const back = () => {
+    return navigation.goBack();
+  };
+  const onPressBack = useCallback(() => {
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/hooks/usePressBack.ts]: where is interval?

File: src/screens/AddNewItemScreen/hooks/useAddNewItemScreen.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import { TodoItemType } from '../../../models';
+
+export type UseAddNewItemScreenParams = {
+  editItem: TodoItemType;
+  isEdit: boolean;
+  date: Date;
+  initialDate: Date;
+};
+
+export const useAddNewItemScreen = (params: UseAddNewItemScreenParams) => {
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/hooks/useAddNewItemScreen.ts:23]: the same

File: src/screens/AddNewItemScreen/index.tsx [hunk 8]
```diff
@@ -1,87 +1,66 @@
...
+  };
+
+  const onConfirm = (date: Date) => {
+    setDate(date);
+    onCancelDate();
+  };
+
+  const onCancelDate = () => {
+    setOpen(false);
+  };
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/index.tsx]: move it to hook, where you hanlde input

File: src/hooks/useTextInput.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { useState } from 'react';
+
+export const useTextInput = (initialValue: string) => {
+  const [value, setText] = useState(initialValue);
+  return {
+    value,
+    onChangeText: setText,
+  };
+};
```
Comment (SHEMPEL133) [inline src/hooks/useTextInput.ts]: it is redundant hook, group it with date state

File: src/screens/AddNewItemScreen/hooks/usePress.ts (hunk 1 above)
Comment (AndrewSakovich) [reply src/screens/AddNewItemScreen/hooks/usePress.ts]: i fix it

File: src/screens/AddNewItemScreen/hooks/usePress.ts (hunk 2 above)
Comment (AndrewSakovich) [reply src/screens/AddNewItemScreen/hooks/usePress.ts]: ok

File: src/screens/AddNewItemScreen/hooks/usePress.ts (hunk 3 above)
Comment (AndrewSakovich) [reply src/screens/AddNewItemScreen/hooks/usePress.ts]: i fix it

File: src/screens/AddNewItemScreen/index.tsx (hunk 4 above)
Comment (AndrewSakovich) [reply src/screens/AddNewItemScreen/index.tsx]: ok

File: src/screens/AddNewItemScreen/hooks/usePressBack.ts (hunk 5 above)
Comment (AndrewSakovich) [reply src/screens/AddNewItemScreen/hooks/usePressBack.ts]: ok

**Review by SHEMPEL133 — APPROVED** [review]

**Pull request merged**